src/expenses.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+
+const express = require('express');
+const router = express.Router();
+
+let expenses = [];
+
+router.get('/', (req, res) => {
+  const { userId, categories, from, to } = req.query;
+
+  let filteredExpenses = expenses;
+
+  if (userId) {
+    filteredExpenses = expenses.filter(exp => exp.userId === +userId);
+  };
+
+  if (categories) {
+    filteredExpenses = expenses
+      .filter(exp => categories.includes(exp.categories));
+  }
+
+  if (from) {
+    filteredExpenses = expenses
+      .filter(exp => new Date(from) <= new Date(exp.spentAt));
+  }
+
+  if (to) {
+    filteredExpenses = expenses
+      .filter(exp => new Date(to) >= new Date(exp.spentAt));
+  }
+
+  res.send(filteredExpenses);
+});
+
+router.post('/', (req, res) => {
+  const { userId, spentAt, title, amount, category, note } = req.body;
+  const newExpenseId = expenses.length + 1;
+
+  const newExp = {
+    id: newExpenseId,
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  };
+
+  if (!userId || !spentAt || !title || !category || !amount || !note) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  expenses.push(newExp);
+  res.sendStatus(201);
+});
+
+router.get('/:expenseId', (req, res) => {
+  const { expenseId } = req.params;
+  const foundExpense = expenses.find(expense => expense.id === +expenseId);
+
+  if (!foundExpense) {
+    res.sendStatus(404);
+
+    return;
```

Comment:
your errors have to be more informative. You want them to be maximally clear, as this makes your code much easier to debug and maintain